xbundle/__init__.py

```diff
@@ -250,7 +250,12 @@ def import_metadata_from_directory(self, dir):
             policies = etree.Element('policies')
             policies.set('semester',os.path.basename(pdir))
             for fn in glob.glob(join(pdir, '*.json')):
-                x = etree.SubElement(policies,os.path.basename(fn).replace('_','').replace('.json',''))
+                try:
+                    x = etree.SubElement(policies,os.path.basename(fn).replace('_','').replace('.json',''))
```

Comment:
Yes, this will still load `myrandom.json`. I think checking for the desired files is correct. However, this is a change to the current functionality; the patch just maintains existing functionality (minus the runtime error).

Yes, the `with` statements would be appropriate. I can add it to a re-work of this patch. I was intentionally making small, focused changes for now so I didn't add confusion (not mixing refactoring with pylint with bug fixes). For example, I intentionally didn't fix this too-long line for ease of code review.